src/components/RoiCalculatorModal/index.tsx

```diff
@@ -0,0 +1,258 @@
+import React, { useRef, useEffect } from 'react'
+import styled from 'styled-components'
+import {
+  Modal,
+  Text,
+  Button,
+  Flex,
+  ButtonMenu,
+  Checkbox,
+  BalanceInput,
+  HelpIcon,
+  ButtonMenuItem,
+  useTooltip,
+} from '@pancakeswap/uikit'
+import BigNumber from 'bignumber.js'
+import { useTranslation } from 'contexts/Localization'
+import { getBalanceNumber } from 'utils/formatBalance'
+import useActiveWeb3React from 'hooks/useActiveWeb3React'
+import RoiCalculatorFooter from './RoiCalculatorFooter'
+import RoiCard from './RoiCard'
+import useRoiCalculatorReducer, { CalculatorMode, EditingCurrency } from './useRoiCalculatorReducer'
+import AnimatedArrow from './AnimatedArrow'
+
+interface RoiCalculatorModalProps {
+  onDismiss?: () => void
+  onBack?: () => void
+  earningTokenPrice: number
+  apr: number
+  displayApr?: string
+  linkLabel: string
+  linkHref: string
+  stakingTokenBalance: BigNumber
+  stakingTokenSymbol: string
+  stakingTokenPrice: number
+  earningTokenSymbol?: string
+  multiplier?: string
+  autoCompoundFrequency?: number
+  performanceFee?: number
+  isFarm?: boolean
+  initialValue?: string
+}
+
+const StyledModal = styled(Modal)`
+  width: 345px;
+  & > :nth-child(2) {
+    padding: 0;
+  }
+`
+
+const ScrollableContainer = styled.div`
+  padding: 24px;
+  max-height: 500px;
+  overflow-y: auto;
+  ${({ theme }) => theme.mediaQueries.sm} {
+    max-height: none;
+  }
+`
+
+const FullWidthButtonMenu = styled(ButtonMenu)<{ disabled?: boolean }>`
+  width: 100%;
+
+  & > button {
+    width: 100%;
+  }
+
+  opacity: ${({ disabled }) => (disabled ? 0.5 : 1)};
+`
+
+const RoiCalculatorModal: React.FC<RoiCalculatorModalProps> = ({
+  onDismiss,
+  onBack,
+  earningTokenPrice,
+  apr,
+  displayApr,
+  linkLabel,
+  linkHref,
+  stakingTokenBalance,
+  stakingTokenSymbol,
+  stakingTokenPrice,
+  multiplier,
+  initialValue,
+  earningTokenSymbol = 'CAKE',
+  autoCompoundFrequency = 0,
+  performanceFee = 0,
+  isFarm = false,
+}) => {
+  const { t } = useTranslation()
+  const { account } = useActiveWeb3React()
+  const balanceInputRef = useRef<HTMLInputElement | null>(null)
+
+  const {
+    state,
+    setPrincipalFromUSDValue,
+    setPrincipalFromTokenValue,
+    setStakingDuration,
+    toggleCompounding,
+    toggleEditingCurrency,
+    setCompoundingFrequency,
+    setCalculatorMode,
+    setTargetRoi,
+  } = useRoiCalculatorReducer(stakingTokenPrice, earningTokenPrice, apr, autoCompoundFrequency, performanceFee)
```

Comment:
Pulling all the data from `useRoiCalculatorReducer` and from now on mostly just displaying them & firing actions. All calculations are in `useRoiCalculatorReducer`